File: finder4j-backtrack/src/main/java/com/github/k0zka/finder4j/backtrack/listener/LoggerSolutionListener.java
package com.github.k0zka.finder4j.backtrack.listener;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.github.k0zka.finder4j.backtrack.SolutionListener;
import com.github.k0zka.finder4j.backtrack.State;
import com.github.k0zka.finder4j.backtrack.Step;

public class LoggerSolutionListener<X extends State, S extends Step<X>> implements SolutionListener<X, S>{

	private  static final Logger logger = LoggerFactory.getLogger(LoggerSolutionListener.class);
	
	private int cntr = 0;
	
	public void onSolution(final X state) {
		logger.info("Solution {} : {}", cntr ++, state);
	}

}

File: finder4j-backtrack/src/main/kotlin/com/github/k0zka/finder4j/backtrack/listener/LoggerSolutionListener.kt
package com.github.k0zka.finder4j.backtrack.listener

import com.github.k0zka.finder4j.backtrack.SolutionListener
import com.github.k0zka.finder4j.backtrack.State
import com.github.k0zka.finder4j.backtrack.Step
import org.slf4j.LoggerFactory

class LoggerSolutionListener<in X : State> : SolutionListener<X> {

	private var cntr = 0

	override fun onSolution(state: X) {
		logger.info("Solution {} : {}", cntr++, state)
	}

	companion object {
		private val logger = LoggerFactory
				.getLogger(LoggerSolutionListener::class.java)
	}

}
